content/docs/features/dockerfiles.md

```diff
@@ -0,0 +1,315 @@
+    same container as) `detect`
+* Invoke the `restorer` with the `-build-image` flag and cache volume mounted at `/kaniko`
+  * The `restorer` will gather data from the registry that is necessary for builder image extension
+* Invoke the `extender` binary with the builder image digest reference as the first argument and cache volume mounted
+  at `/kaniko`
+  * Note that when extending the builder image, there is no need to invoke the `builder` binary as the `build` phase is
+    a sub-task of the `extender` and thus happens automatically after (and in the same container as) `extend`
+* Invoke the `exporter` as usual
+
+Extensions workflows are not currently supported when using the `creator` binary - support may be added in the future,
+but with a lower priority.
+
+### Risks
+
+Image extensions are considered experimental and susceptible to change in future API versions. Additionally, platform
+operators should be mindful that:
+
+* Dockerfiles are very powerful - in fact, you can do anything with a Dockerfile! Introducing image extensions into your
+  CNB builds can eliminate the security and compatibility guarantees that buildpacks provide if not done with great
+  care.
+* When Dockerfiles are used to switch the run image from that defined on the provided builder, the resulting run image
+  may not have all the mixins required by buildpacks that detected. Platforms may wish to optionally re-validate mixins
+  prior to `build` when using extensions.
+
+### Phased approach
+
+Some limitations of the initial implementation of the Dockerfiles feature have already been mentioned, and we'll expand
+on them here. As this is a large and complicated feature, the implementation has been split into phases in order to
+deliver incremental value and gather feedback.
+
+* Phase 1 (supported in lifecycle `0.15.0` or greater): one or more `run.Dockerfile`s each containing a single `FROM`
+  instruction can be used to switch the original run image to a new image (as no image modifications are permitted,
+  there is no need to run `extend` on the run image)
+* Phase 2 (supported in lifecycle `0.15.0` or greater): one or more `build.Dockerfile`s can be used to extend the
+  builder image
+  * A new `extend` lifecycle phase is introduced to apply `build.Dockerfile`s from `generate` to the builder image
+* Phase 3 (future): one or more `run.Dockerfile`s can be used to extend the run image
+  * The `extend` lifecycle phase can be run in parallel for the builder and run images
+
+The final ordering of lifecycle phases will look something like the following:
+
+* `analyze`
+* `detect` - after standard detection, `detect` will also run extensions' `./bin/generate`; output Dockerfiles are
+  written to a volume
+* `restore`
+* `extend` - applies one or more `build.Dockerfile`s to the builder image
+* `extend` - applies one or more `run.Dockerfile`s to the run image (could run in parallel with builder image extension)
+* `build`
+* `export`
+
```

Comment:
Should we have a subsection here of "Platform Support"?

Reply:
I'm adding a migration guide for 0.9 -> 0.10 and a link to that in this section.